sparkpost/templates.py

```diff
@@ -0,0 +1,182 @@
+            Used to track clicks of transmission
+        :param bool is_transactional: Defaults to transmission level setting.
+            Distinguishes between transactional and non-transactional messages
+            for unsubscribe and suppression purposes
+
+        :param str html: HTML part of template
+        :param str text: Text part of template
+        :param str subject: Subject of template
+        :param str from_email: Friendly from of template, domain must be a
+            verified sending domain to your account or template create
+            will fail
+        :param str reply_to: Reply to of template
+        :param dict custom_headers: Used to set any headers associated with
+            template
+
+        :returns: TODO
+        :raises: :exc:`SparkPostAPIException` if template is not found
+        """
+        uri = "%s/%s" % (self.uri, template_id)
+        payload = self._translate_keys(**kwargs)
+        results = self.request('PUT', uri, data=json.dumps(payload))
+        return results
+
+    def delete(self, template_id):
+        """
+        Delete a template by ID
+
+        :param str template_id: ID of the template you want to delete
+
+        :returns: TODO
+        :raises: :exc:`SparkPostAPIException` if template is not found or if
+        template is in use
+        """
+        uri = "%s/%s" % (self.uri, template_id)
+        results = self.request('DELETE', uri)
+        return results
+
+    def get(self, template_id, is_draft=None):
+        """
+        Get a template by ID
+
+        :param str template_id: ID of the template you want to retrieve
+        :param bool is_draft: Defaults to None. If True, returns the most
+            recent draft template. If False, returns the most recent published
+            template. If None, returns the most recent template version
+            regardless of draft or published.
+
+        :returns: the requested template if found
+        :raises: :exc:`SparkPostAPIException` if template is not found
+        """
+        uri = "%s/%s" % (self.uri, template_id)
+        params = {}
+        if is_draft is not None:
+            params['draft'] = str(is_draft).lower()
+        results = self.request('GET', uri, params=params)
+        return results
+
+    def list(self):
+        """
+        Get a list of your templates
+
+        :returns: list of templates
+        :raises: :exc:`SparkPostAPIException` if API call fails
+        """
+        results = self.request('GET', self.uri)
+        return results
+
+    def preview(self, template_id, substitution_data, is_draft=None):
+        """
+        Get a preivew of your template by ID with the
+        provided substitution_data
+
+        :param str template_id: ID of the template you want to retrieve
+        :param dict substitution_data: data to be substituted in the
+            template content
+        :param bool is_draft: Defaults to None. If True, previews the most
```

Comment:
OK just ignore me - I was confusing this with the published param for the ``create`` method.